flink-runtime/src/main/scala/org/apache/flink/runtime/taskmanager/TaskManager.scala

```diff
@@ -417,6 +417,28 @@ class TaskManager(
             log.debug(s"Cannot find task to fail for execution $executionID)")
           }
 
+        // stops a task
+        case StopTask(executionID) =>
+          val task = runningTasks.get(executionID)
+          if (task != null) {
+            try {
+              task.stopExecution()
+              sender ! decorateMessage(new TaskOperationResult(executionID, true))
+            } catch {
+              case t: Throwable =>
+                        sender ! new TaskOperationResult(executionID, false,
```

Comment:
Sending for every `Throwable` a `TaskOperationResult(.., false, ...)` back will result in a logging statement on the `JobManager` side that the corresponding task could not be found. This seems to be wrong since the task was found but an exception occurred while calling stop on it. The `stop` method of `Task` can throw an `UnsupportedOperationException` which should lead on the `JobManager` side to a logging statement `Task does not support stop`. Every other exception should lead to a cancellation of the corresponding `Task`, because something obviously went wrong.